File: java-algorithm/src/main/java/com/leetcode/Problem804.java
package com.leetcode;

import java.util.HashSet;
import java.util.Set;

/**
 * 804. Unique Morse Code Words
 * https://leetcode.com/problems/unique-morse-code-words/
 */
public class Problem804 {
    public int uniqueMorseRepresentations(String[] words) {
        String[] codes = new String[]{".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."};
        Set<String> set = new HashSet<>();
        for (String word : words) {
            StringBuffer sb = new StringBuffer();
            for (char ch : word.toCharArray()) {
                sb.append(codes[ch - 'a']);
            }
            String tmp = sb.toString();
            set.add(tmp);
        }
        return set.size();
    }
}

File: python-algorithm/leetcode/problem_804.py
"""804. Unique Morse Code Words
https://leetcode.com/problems/unique-morse-code-words
"""
from typing import List


class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        codes = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        store = set()
        for word in words:
            cur = ''
            for ch in word:
                cur += codes[ord(ch) - 97]
            store.add(cur)
        return len(store)
